### PR TITLE
soc: nordic_nrf: Add a couple of missing nRF528xx variants

### PR DESCRIPTION
**Pull request by carlescufi**
Add:

- nRF52840 QFAA
- nRF52833 QDAA

### CODE REVIEW

File: dts/arm/nordic/nrf52833_qdaa.dtsi [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	reg = <0x00000000 DT_SIZE_K(512)>;
+};
+
+&sram0 {
+	reg = <0x20000000 DT_SIZE_K(128)>;
+};
+
+/ {
+	soc {
+		compatible = "nordic,nRF52833-QDAA", "nordic,nRF52833", "nordic,nRF52", "simple-bus";
```
Comment (gmarull) [inline dts/arm/nordic/nrf52833_qdaa.dtsi:20]:
```suggestion
		compatible = "nordic,nrf52833-qdaa", "nordic,nrf52833", "nordic,nrf52", "simple-bus";
```

File: dts/arm/nordic/nrf52840_qfaa.dtsi [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	reg = <0x00000000 DT_SIZE_K(1024)>;
+};
+
+&sram0 {
+	reg = <0x20000000 DT_SIZE_K(256)>;
+};
+
+/ {
+	soc {
+		compatible = "nordic,nRF52840-QFAA", "nordic,nRF52840", "nordic,nRF52", "simple-bus";
```
Comment (gmarull) [inline dts/arm/nordic/nrf52840_qfaa.dtsi:20]:
```suggestion
		compatible = "nordic,nrf52840-qfaa", "nordic,nrf52840", "nordic,nrf52", "simple-bus";
```

File: dts/arm/nordic/nrf52833_qdaa.dtsi (hunk 1 above)
Comment (carlescufi) [reply dts/arm/nordic/nrf52833_qdaa.dtsi:20]: Are you sure? that's not how we have it in the other SoCs today

Comment (gmarull) [reply dts/arm/nordic/nrf52833_qdaa.dtsi:20]: I've never seen compatibles using uppercase, but I don't see any rule in the dt spec

File: dts/arm/nordic/nrf52840_qfaa.dtsi [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	reg = <0x00000000 DT_SIZE_K(1024)>;
+};
+
+&sram0 {
+	reg = <0x20000000 DT_SIZE_K(256)>;
+};
+
+/ {
+	soc {
+		compatible = "nordic,nRF52840-QFAA", "nordic,nRF52840", \
```
Comment (gmarull) [inline dts/arm/nordic/nrf52840_qfaa.dtsi:20]: is `\` needed?

Comment (carlescufi) [reply dts/arm/nordic/nrf52840_qfaa.dtsi:20]: Not sure, do you know?

**Review by anangl — COMMENTED** [review]
Since the High Voltage supply mode cannot be used in nRF52840-QFAA, I think we should change the dependency in https://github.com/zephyrproject-rtos/zephyr/blob/d7c00ee7c25b48ca31ee6702dc6516786d98963a/soc/arm/nordic_nrf/nrf52/Kconfig.soc#L82-L84
to `SOC_NRF52840_QIAA`.

File: dts/arm/nordic/nrf52840_qfaa.dtsi (hunk 3 above)
Comment (anangl) [reply dts/arm/nordic/nrf52840_qfaa.dtsi:20]: It's not needed.

File: dts/arm/nordic/nrf52833_qdaa.dtsi (hunk 1 above)
Comment (carlescufi) [reply dts/arm/nordic/nrf52833_qdaa.dtsi]: @anangl your opinion here?

File: dts/arm/nordic/nrf52840_qfaa.dtsi (hunk 3 above)
Comment (carlescufi) [reply dts/arm/nordic/nrf52840_qfaa.dtsi:20]: Thanks both, will remove!

File: dts/arm/nordic/nrf52833_qdaa.dtsi (hunk 1 above)
Comment (anangl) [reply dts/arm/nordic/nrf52833_qdaa.dtsi]: In Devicetree Specification v0.4 (released 28 June) the following note was added in section "2.3.1 compatible":
> The compatible string should consist only of lowercase letters, digits and dashes, and should start with a letter.
A single comma is typically only used following a vendor prefix. Underscores should not be used.

I think we should correct those compatibles in all our SoCs then.

File: dts/arm/nordic/nrf52833_qdaa.dtsi [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+&sram0 {
+	reg = <0x20000000 DT_SIZE_K(128)>;
+};
+
+/ {
+	soc {
+		compatible = "nordic,nRF52833-QDAA", "nordic,nRF52833",
+			     "nordic,nRF52", "simple-bus";
```
Comment (gmarull) [inline dts/arm/nordic/nrf52833_qdaa.dtsi:21]: will this be corrected later then?

Comment (carlescufi) [reply dts/arm/nordic/nrf52833_qdaa.dtsi:21]: Yes, I will send a follow-up PR today.

**Review by anangl — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**